ui/layout.js

```diff
@@ -116,19 +117,63 @@ const bgGroups = [
 
 var OverviewCloneController = class OverviewCloneController {
     constructor() {
+        this._overviewShowingId = 0;
+        this._overviewShownId = 0;
+        this._overviewHidingId = 0;
+        this._overviewHiddenId = 0;
+        this._viewSelectorPageChangedId = 0;
+    }
+
+    _updateClones() {
+        const { visible, animationInProgress } = Main.overview;
+        const activePage = Main.overview.viewSelector.getActivePage();
+        const inWindowsPage = activePage === ViewSelector.ViewPage.WINDOWS;
+
+        const overviewCloneOpacity =
+            animationInProgress || inWindowsPage ? 255 : 0;
+        Main.overview._backgroundGroup._appGridClone.opacity = overviewCloneOpacity;
+
+        const layoutCloneOpacity = visible ? 0 : 255;
+        Main.layoutManager._backgroundGroup._appGridClone.opacity = layoutCloneOpacity;
     }
 
     enable() {
-        bgGroups.forEach(group => group.add_child(new OverviewClone()));
+        bgGroups.forEach(group => {
+            const clone = new OverviewClone();
+
+            group.add_child(clone);
+            group._appGridClone = clone;
+        });
+
+        this._overviewShowingId =
+            Main.overview.connect('showing', () => this._updateClones());
+        this._overviewShownId =
+            Main.overview.connect('shown', () => this._updateClones());
+        this._overviewHidingId =
+            Main.overview.connect('hiding', () => this._updateClones());
+        this._overviewHiddenId =
+            Main.overview.connect('hidden', () => this._updateClones());
+        this._viewSelectorPageChangedId =
+            Main.overview.viewSelector.connect('page-changed',
+                () => this._updateClones());
     }
 
     disable() {
         bgGroups.forEach(actor => {
-            for (const child of actor) {
-                if (child instanceof OverviewClone)
-                    child.destroy();
-            }
+            if (actor._appGridClone)
+                actor._appGridClone.destroy();
```

Comment:
Should you also `delete actor._appGridClone` (or at least set it to null) here? Also not required for this current code but in case you start checking if the clone exists in the group it may bring issues.